### PR TITLE
fix EOL workflow for pending major releases

### PR DESCRIPTION
**Pull request by szaimen**

### CODE REVIEW

File: .github/workflows/block-merge-eol.yml [hunk 1]
```diff
@@ -27,4 +27,11 @@ jobs:
  
       - name: Checking if ${{ env.server_major }} is EOL
         run: |
-          php -r 'echo json_encode(require_once "config.php");' | jq --arg version "${{ env.server_major }}" '.stable[$version]["100"].eol' | grep --silent -i 'false'
+          if php -r 'echo json_encode(require_once "config.php");' | jq --arg version "${{ env.server_major }}" '.stable[$version]["100"].eol' | grep --silent -i 'false'; then
+            exit 0
+          fi
+
+          PREVIOUS_MAJOR="$((env.server_major-1))"
```
Comment (skjnldsv) [inline .github/workflows/block-merge-eol.yml:34]: This does not seem compliant to the way github workflow variables works? 
`env.server_major` would not be transformed here without `${{ env.server_major }}`, no?

Comment (szaimen) [reply .github/workflows/block-merge-eol.yml]: right, I've solved it by making a copy into a usual variable.

**Comment by szaimen** [comment]
> Instead we could check if beta has it covered?

done

**Comment by max-nextcloud** [comment]
> > Instead we could check if beta has it covered?
> 
> done

I think if you check beta you don't need to look at the previous release, right? For example stable26 -> 26 is in beta right now. We don't have a stable27 branch yet.

**Comment by szaimen** [comment]
> I think if you check beta you don't need to look at the previous release, right? For example stable26 -> 26 is in beta right now. We don't have a stable27 branch yet.

right, my mistake

**Comment by szaimen** [comment]
> > I think if you check beta you don't need to look at the previous release, right? For example stable26 -> 26 is in beta right now. We don't have a stable27 branch yet.
> 
> right, my mistake

fixed

**Review by max-nextcloud — APPROVED** [review]

**Review by blizzz — CHANGES_REQUESTED** [review]
please open and complete against https://github.com/nextcloud/.github first

**Comment by szaimen** [comment]
> please open and complete against https://github.com/nextcloud/.github first

done in https://github.com/nextcloud/.github/pull/130

**Review by skjnldsv — APPROVED** [review]

File: .github/workflows/block-merge-eol.yml [hunk 2]
```diff
@@ -27,4 +27,10 @@ jobs:
...
       - name: Checking if ${{ env.server_major }} is EOL
         run: |
-          php -r 'echo json_encode(require_once "config.php");' | jq --arg version "${{ env.server_major }}" '.stable[$version]["100"].eol' | grep --silent -i 'false'
+          if php -r 'echo json_encode(require_once "config.php");' | jq --arg version "${{ env.server_major }}" '.stable[$version]["100"].eol' | grep --silent -i 'false'; then
+            exit 0
+          fi
+
+          if ! php -r 'echo json_encode(require_once "config.php");' | jq --arg version "${{ env.server_major }}" '.beta[$version]["100"].eol' | grep --silent -i 'false'; then
+            exit 1
+          fi
```
Comment (skjnldsv) [inline .github/workflows/block-merge-eol.yml:36]: You can use fallback in jq: https://stedolan.github.io/jq/manual/#Alternativeoperator://
```suggestion
          if php -r 'echo json_encode(require_once "config.php");' | jq --arg version "${{ env.server_major }}" '.stable[$version]["100"].eol // .beta[$version]["100"].eol'
' | grep --silent -i 'false'; then
            exit 0
          fi
```

**Review by skjnldsv — CHANGES_REQUESTED** [review]

**Pull request closed without merging**